### PR TITLE
fix: move testId to inputProps for all input components

### PR DESCRIPTION
**Pull request by trevoring-okta**
## Summary
- Existing implementation of `testId` had placed it on the wrapper MUI element for all MUI components that wrap an underlying raw HTML `input` element (e.g. `TextField`, `Checkbox`, `Radio`, etc.). Most downstream repos have written their unit and e2e tests expecting that the `data-se` attribute is on the raw HTML `input` element rather than a wrapper parent element, so this PR moves `data-se` to MUI's `inputProps`
    - Required module augmentation of React's `HTMLAttributes` to allow `data-*` attributes because of an underlying issue with TypeScript
    - Moved `FocusHandle` out of declaration file because it fits better in `inputUtils.ts`
    - Fixes missing input focus ref and autocomplete props from `NativeSelect`

### CODE REVIEW

**Review by bryancunningham-okta — DISMISSED** [review]
This seems fine to me. Asking for @chrispulsinelli-okta for some help getting an approval for this

File: packages/odyssey-react-mui/src/@types/react-augment.d.ts
Comment (chrispulsinelli-okta) [inline packages/odyssey-react-mui/src/@types/react-augment.d.ts:13]: Any reason why you removed `FC` as an imported object? I believe this affects the interface definition below where it extends `FC`

File: packages/odyssey-react-mui/src/labs/VirtualizedAutocomplete.tsx
Comment (trevoring-okta) [inline packages/odyssey-react-mui/src/labs/VirtualizedAutocomplete.tsx:279]: lint fix here @bryancunningham-okta @chrispulsinelli-okta

File: packages/odyssey-react-mui/src/@types/react-augment.d.ts
Comment (chrispulsinelli-okta) [reply packages/odyssey-react-mui/src/@types/react-augment.d.ts:13]: I believe this should be something like `import React, { FC } from "react";`

**Review by chrispulsinelli-okta — DISMISSED** [review]
I'm OK with this change and leaving an approval as to not block you, but please review and update, if necessary, the change to the import statement in `react-augment.d.ts` please. Once that is addressed, this should be fine to merge.

**Comment by chrispulsinelli-okta** [comment]
You have the following errors in the build:
```
Error: src/labs/DatePicker.tsx(21,10): error TS2590: Expression produces a union type that is too complex to represent.
lerna ERR! yarn run typecheck exited 2 in '@okta/odyssey-react-mui'
```

File: packages/odyssey-react-mui/src/@types/react-augment.d.ts
Comment (trevoring-okta) [reply packages/odyssey-react-mui/src/@types/react-augment.d.ts:13]: @chrispulsinelli-okta So this change to `react-augment.d.ts` came from a discussion with Kevin. I saw that you were the original author of this file but Kevin was saying he thought the `ForwardRefWithType` definition would actually fit better in `inputUtils.ts` (just like what I did with `FocusHandle`) since [`d.ts` files shouldn't be used for holding custom types](https://github.com/microsoft/TypeScript/issues/52593#issuecomment-1419505081). I'm not a TS expert so I'm most relaying what Kevin was saying, but he originally wanted me to move your `ForwardRefWithType` interface into `inputUtils.ts` in this ticket as well but it led to a bunch of TS compiler errors when I tried to do so since Odyssey has `"skipLibCheck": true` in `tsconfig` which prevents type checking in `d.ts` files. 

I don't have any context behind `ForwardRefWithType` and didn't want to break your code so Kevin asked me to create a ticket and assign it to you for this sprint: https://oktainc.atlassian.net/browse/OKTA-688308

Regardless, I can still add back the `FC` for the scope of this PR

**Comment by trevoring-okta** [comment]
> You have the following errors in the build:
> 
> ```
> Error: src/labs/DatePicker.tsx(21,10): error TS2590: Expression produces a union type that is too complex to represent.
> lerna ERR! yarn run typecheck exited 2 in '@okta/odyssey-react-mui'
> ```

@chrispulsinelli-okta I didn't touch the `DatePicker` component as part of this PR but I can still try to resolve the issue. Do you anticipate this blocking the publishing of this PR if it doesn't get resolved?

File: packages/odyssey-react-mui/src/NativeSelect.tsx
Comment (trevoring-okta) [inline packages/odyssey-react-mui/src/NativeSelect.tsx:191]: @KevinGhadyani-Okta

File: packages/odyssey-react-mui/src/Select.tsx
Comment (trevoring-okta) [inline packages/odyssey-react-mui/src/Select.tsx:294]: @KevinGhadyani-Okta

File: packages/odyssey-react-mui/src/@types/react-augment.d.ts
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/@types/react-augment.d.ts:22]: I will approve, but please check if the `type` definition at line 20 is supposed to go inside the `declare` at line 21.

I didn't notice it before, but I'm thinking that's how you localize global type overrides.

File: packages/odyssey-react-mui/src/NativeSelect.tsx
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/NativeSelect.tsx:49]: Nice documentation comment! 👍

File: packages/odyssey-react-mui/src/NativeSelect.tsx
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/NativeSelect.tsx:121]: I'd like this to be more generic.

Can we rename this `inputRef` so maybe `.blur()` is possible in the future and others?

The internal one could be named `localInputRef` to differentiate it.

File: packages/odyssey-react-mui/src/NativeSelect.tsx
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/NativeSelect.tsx:151]: I'd like to have this match other places in the code:
```suggestion
            element?.focus();
```

File: packages/odyssey-react-mui/src/NativeSelect.tsx
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/NativeSelect.tsx:148]: I don't think we need to create this `element` variable.

I'm fine having `inputRef.current?.focus()` below.

**Review by KevinGhadyani-Okta — COMMENTED** [review]
A few comments, but it's otherwise very close! :)

**Review by KevinGhadyani-Okta — APPROVED** [review]

**Pull request merged**
